pandas/io/json/json.py

```diff
@@ -349,6 +386,13 @@ def read_json(path_or_buf=None, orient=None, typ='frame', dtype=True,
         lines = list(StringIO(json.strip()))
         json = '[' + ','.join(lines) + ']'
 
+    return _get_obj(typ, json, orient, dtype, convert_axes, convert_dates,
+                    keep_default_dates, numpy, precise_float, date_unit)
+
+
+def _get_obj(typ, json, orient, dtype, convert_axes, convert_dates,
+             keep_default_dates, numpy, precise_float,
+             date_unit):
```

Comment:
Before, this code was only called once. Now it needs to be called from two places, so I extracted it into a function (see line 343). 

One possible improvement I was considering is to define the function inline in the read_json function so that we don't need to pass so many parameters.